public/src/modules/colors.js

```diff
@@ -0,0 +1,30 @@
+// const colors = {
+//     "hover_select": hover_select,
+// }
+
+const getColors = () => {
```

Comment:
This function is called in many places. This function extracts the property of each style from the document model right? This is not very efficient and can cause performance issues. Store this in an object and reference this object everywhere instead? Maybe some thing like have a globalObject like `selectedColorTheme` and do `selectedColorTheme = getColors()` when theme changes. Then everywhere else we can refer colors by `selectedColorTheme['color-code'].

Reply:
Yes, this will be better, I will make the changes.